src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs

```diff
@@ -15,23 +17,45 @@ namespace System.Net.Security.Tests
 
     public class CertificateValidationRemoteServer
     {
-        [Fact]
         [OuterLoop("Uses external servers")]
-        public async Task CertificateValidationRemoteServer_EndToEnd_Ok()
+        [ConditionalTheory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task CertificateValidationRemoteServer_EndToEnd_Ok(bool useAsync)
         {
-            if (PlatformDetection.IsWindows7)
-            {
-                // https://github.com/dotnet/corefx/issues/42339
-                return;
-            }
-
             using (var client = new TcpClient(AddressFamily.InterNetwork))
             {
-                await client.ConnectAsync(Configuration.Security.TlsServer.IdnHost, Configuration.Security.TlsServer.Port);
+                try
+                {
+                    await client.ConnectAsync(Configuration.Security.TlsServer.IdnHost, Configuration.Security.TlsServer.Port);
+                }
+                catch (Exception ex)
+                {
+                    // if we cannot connect, skip the test instead of failing.
+                    // This test is not trying to test networking.
+                    new SkipTestException($"Unable to connect to '{Configuration.Security.TlsServer.IdnHost}': {ex.Message}");
+                }
 
                 using (SslStream sslStream = new SslStream(client.GetStream(), false, RemoteHttpsCertValidation, null))
                 {
-                    await sslStream.AuthenticateAsClientAsync(Configuration.Security.TlsServer.IdnHost);
+                    try
+                    {
+                        if (useAsync)
+                        {
+                            await sslStream.AuthenticateAsClientAsync(Configuration.Security.TlsServer.IdnHost);
+                        }
+                        else
+                        {
+                            sslStream.AuthenticateAsClient(Configuration.Security.TlsServer.IdnHost);
+                        }
+                    }
+                    catch (IOException ex) when (ex.InnerException is SocketException &&
+                      ((SocketException)ex.InnerException).SocketErrorCode == SocketError.ConnectionReset)
+                    {
+                        // Since we try to verify certificate validation, ignore IO errors
+                        // caused most likely by environmental failures.
+                        new SkipTestException($"Unable to connect to '{Configuration.Security.TlsServer.IdnHost}': {ex.InnerException.Message}");
```

Comment:
I think this should be `throw new SkipTestException()'.  Doing just `new SkipTestException()` won't do anything.